### ISSUE TITLE
beginend support breaks several features

### ISSUE THREAD

**Issue by leungbk** [emacs-evil/evil-collection#533]
1. `beginend-foo-goto-beginning` and `-end` don't presently have the `:jump` property set to `t`.
2. In Vim, pressing `123G` should take me to line 123. The `beginend-foo-goto-end` commands don't support this. We might have to make some wrappers.

cc @balajisivaraman

**Comment by balajisivaraman** [comment]
@leungbk, Thanks for pointing this out. I'm working on a fix for it. (I didn't know that this motion existed either in Vim or Evil, since I have never used it.)

**Comment by balajisivaraman** [comment]
@leungbk, Does this look like an acceptable fix?

```emacs-lisp
(evil-define-motion evil-beginend-prog-mode-goto-beginning (count)
  :jump t
  :type line
  (if count
      (evil-goto-first-line count)
    (beginend-prog-mode-goto-beginning)))

(evil-define-motion evil-beginend-prog-mode-goto-end (count)
  :jump t
  :type line
  (if count
      (evil-goto-line count)
    (beginend-prog-mode-goto-end)))kk
```

It works when I tried it locally. I will have to use a macro probably to do this for all the `beginend-foo-goto-beginning` functions.

I originally tried just doing `(evil-add-command-properties 'beginend-foo-goto-beginning :jump t)` and `(evil-set-command-property 'beginend-foo-goto-beginning :jump t)`, but this did not work. This is why I had to resort to the above solution of defining a new motion, which is working.

If this is acceptable, I'll work on a proper fix for all the beginend modes. Cheers.
